pkg/cli/cli.go

```diff
@@ -207,6 +273,26 @@ func (c cli) validate() error {
 						pluginName, projectVersion, err)
 				}
 			}
+			// Check for duplicate plugin names. Names outside of a version can
+			// conflict because multiple project versions of a plugin may exist.
+			if _, seen := pluginNameSet[pluginName]; seen {
+				return fmt.Errorf("two plugins have the same name: %q", pluginName)
+			}
+			pluginNameSet[pluginName] = struct{}{}
+		}
+	}
+
+	// Validate plugin keys set in CLI.
+	for pluginName, pluginVersion := range c.cliPluginKeys {
```

Comment:
IHMO the impl could be:
- parseFlags()
- validate()
- save()

I mean the plugin validations should be encapsulated in a Validate method and not be in the parseFlags.

Reply:
This is actually in `func (c cli) validate() error`. GitHub hides some lines and I think you didn't opened them.